### ISSUE TITLE
Batching KMS requests

### ISSUE THREAD

**Issue by tombentley** [kroxylicious/kroxylicious#976]
Some KMSes support making batched requests (e.g. HashiCorp for [DEK decryption](https://developer.hashicorp.com/vault/api-docs/secret/transit#decrypt-data)). Right now our `Kms` interface prevents a `Kms` implementation from using batched requests, e.g:

```java
CompletionStage<SecretKey> decryptEdek(@NonNull E edek)
```

We should consider having `Kms` support batched requests.

**Comment by tombentley** [comment]
Note Caffeine support batched based loading, which we could making use of before we make this change, but they'd work better together.

**Comment by k-wall** [comment]
Yes, I agree this is a useful optimization.   I'd noticed that the Vault API supports this use-case just last week.
